ga4gh/repo_manager.py

```diff
@@ -0,0 +1,383 @@
+"""
+Manages a GA4GH data repository
+"""
+# TODO
+# - options for json metas
+# - options for symlink / copy
+# - indexing
+# - remove functions
+# - tests
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+import os
+
+
+def filenameExtension(filename):
+    # os.path.splitext(filename)[1] messes up
+    # on filenames with more than one period
+    return '.' + '.'.join(filename.split('.')[1:])
+
+def filenameWithoutExtension(filename):
+    # os.path.splitext(filename)[0] messes up
+    # on filenames with more than one period
+    return filename.split('.')[0]
+
+
+class AbstractRepoObject(object):
```

Comment:
I'm not sure I see the value of having a separate RepoObject hierarchy. Wouldn't it be simpler to include the required functionality into the ``datamodel.DatamodelObject`` hierarchy? I guess most of the functionality belongs in the PysamDatamodelMixin?